packages/types/message.d.ts

```diff
@@ -0,0 +1,19 @@
+export interface CumulusMeta {
+  execution_name: string
+  queueName?: string
+  state_machine: string
+  parentExecutionArn?: string
+  asyncOperationId?: string
+}
+
+export interface Meta {
+  workflow_name: string
+  collection?: object
```

Comment:
Should we consider going further into this object and start replicating schema definitions?

Reply:
I suppose that can only be a good thing? Having a type for a collection, provider, etc could only improve the value of our type safety? @yjpa7145 ?

Reply:
> Should we consider going further into this object and start replicating schema definitions?

I'm inclined to say "no", at least not here. For example, I do think that we should have a `provider` type that's related to the model schema, but I think it should live in (and be exported by) the API package. We should also probably have a `granule` type that correlates to the output schema of the `discoverGranules` task, but that should live in the `discoverGranules` package.

The thing I want to be careful of is that, in different contexts (certain tasks, packages, etc), terms like "granule" and "collection" have different meanings. A collection in one context may only have a name and a version. In another context, it may be a full collection record as returned by the database.

As I'm typing this, I wonder if this `message` type that we're defining here would actually belong in the `message` package. Thoughts?

Reply:
I think the reason I put the message types in here was because I thought @Jkovarik might need them for the CMA JS work he's going to do. It also seems like we use a Cumulus message as an input to things all over the place (like the API package lambdas, etc), so having the type for it available from a package seemed ideal.

I can see your point about "provider" objects with different shapes in different contexts, but would it still not be good to have one that matches the schema definition for the Cumulus message? That way people are getting a precise type expectation for the provider they pull off of the message. I'll hold off on that for now though

Reply:
> I think the reason I put the message types in here was because I thought @Jkovarik might need them for the CMA JS work he's going to do. It also seems like we use a Cumulus message as an input to things all over the place

I completely agree that we should have a common CumulusMessage type that we pass around. I was just suggesting that the `@cumulus/message` package might be the place for it to be exported from.

Reply:
> I can see your point about "provider" objects with different shapes in different contexts, but would it still not be good to have one that matches the schema definition for the Cumulus message?

Is there a standard for what a provider looks like in a Cumulus message? My understanding is that the provider in the message is just whatever the individual tasks need. (That being said, it probably _is_ the same everywhere.) Let's talk about this in-person.

Reply:
@markdboyd  re: my work, "object" works, it's just less typesafe.   More than we currently have though.   

> Is there a standard for what a provider looks like in a Cumulus message?

@yjpa7145  I would expect that would be the Collection schema, right?

Reply:
@yjpa7145  

> Is there a standard for what a provider looks like in a Cumulus message? My understanding is that the provider in the message is just whatever the individual tasks need. (That being said, it probably is the same everywhere.) Let's talk about this in-person.

Your point is a good one -- nothing forces the 'collection' to be schema-complete for any and all tasks, the only way that's enforced is via task schemas. 

That said, given 30 seconds of thought, it sounds right-direction-ish to more formally couple our Collection type to map to the db schema, possibly with relaxed required fields.